animalcontrol.lua

```diff
@@ -0,0 +1,213 @@
+            ,"str","agi","tgh","endur","recup","disres"))
+    else
+        print(string.format("%-20s %-6s %-9s %-9s %-5s %-22s %-8s %-25s"
+            , "animal type", "id", "unit id", "race id", "sex", "marked for slaughter", "gelded", "marked for gelding"))
+    end
+    for k,v in safe_pairs(Units) do
+        if v.civ_id == df.global.ui.civ_id and v.flags1.tame then
+            if not (args.male or args.female) or args.male and v.sex == 1 or args.female and v.sex == 0 then
+                if not args.race or tonumber(args.race) == v.race then
+                    if not args.markedfor or (args.markedfor == "slaughter" and v.flags2.slaughter) or (args.markedfor == "gelding" and v.flags3.marked_for_gelding) then
+                        if not args.notmarkedfor or (args.notmarkedfor == "slaughter" and not v.flags2.slaughter) or (args.notmarkedfor == "gelding" and not v.flags3.marked_for_gelding) then
+                            if not args.gelded or v.flags3.gelded then
+                                if not args.notgelded or not v.flags3.gelded then
+                                    if not args.id or tonumber(args.id) == v.id then
+                                        count = count + 1
+                                        name = dfhack.units.isAdult(v) and df.global.world.raws.creatures.all[v.race].name[0] or dfhack.units.getRaceChildName(v)
+                                        sex = v.sex == 1 and "M" or "F"
+                                        if args.markfor or args.unmarkfor then
+                                            if args.markfor then
+                                                mark = args.markfor
+                                                state = true
+                                            else
+                                                mark = args.unmarkfor
+                                                state = false
+                                            end
+
+                                            if mark == "gelding" and sex == "M" then
+                                                --print("geld",state)
+                                                v.flags3.marked_for_gelding = state
+                                            elseif mark == "slaughter" then
+                                                --print("slaughter",state)
+                                                v.flags2.slaughter = state
+                                            end
+                                        end
+                                        attr = v.body.physical_attrs
+                                        if v.sex == 1 then
+                                            if args.showstats then
+                                                print(string.format("%-20s %-6s %-9d %-9d %-5s %-22s %-8s %-25s %-7d %-7d %-7d %-7d %-7d %-7d"
+                                                    ,name,v.id,k,v.race,sex
```

Comment:
By the way, the `unit id` column you're including here is actually the index of that unit in world.units.active, which is not the same as the unit ID (your `id` column)